utils/hash_utils/src/key.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::Hash;
+use serde::{Deserialize, Serialize};
+use std::borrow::Borrow;
+use std::hash::Hasher;
+use std::ops::Deref;
+
+/// Key type to be used to serialize as byte string instead of a `u8` array.
+/// This type is used as a default for the `Hamt` as this is the only allowed type
+/// with the go implementation.
+#[derive(Eq, PartialOrd, Clone, Debug, Serialize, Deserialize)]
+#[serde(transparent)]
+pub struct BytesKey(#[serde(with = "serde_bytes")] pub Vec<u8>);
+
+impl PartialEq for BytesKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
```

Comment:
Can't this be derived?

Reply:
Well iirc it was because clippy warns against manually impl `Hash` and deriving `PartialEq`. Or maybe this isn't an issue now that it's our own hash trait. I'll try switching

Reply:
Ohhh wait sorry no it can't because it's own own trait. Reason for our own trait is because of issues like: https://github.com/rust-lang/rust/issues/27108.